pandas-stubs/core/indexes/datetimes.pyi

```diff
@@ -102,11 +102,48 @@ class DatetimeIndex(
         self, periods: int = 1, freq: DateOffset | Timedelta | str | None = None
     ) -> Self: ...
 
+@overload
 def date_range(
-    start: str | DateAndDatetimeLike | None = None,
-    end: str | DateAndDatetimeLike | None = None,
-    periods: int | None = None,
-    freq: str | timedelta | Timedelta | BaseOffset = "D",
+    *,
+    end: str | DateAndDatetimeLike = ...,
+    periods: int = ...,
```

Comment:
I think this would allow `pd.date_range()` to be accepted.

The error message is " Of the four parameters: start, end, periods, and freq, exactly three must be specified".

So if you want this to work, you have to have overloads where 3 are required.  You can't have `= ...` on any of them (or default values either), because that means the argument is optional.

Reply:
If I look at the docs, they don't even use three exactly, when looking at the code they are using the fact that freq defaults to None but is then replaced by D for day.
https://pandas.pydata.org/docs/dev/reference/api/pandas.timedelta_range.html#pandas.timedelta_range

See https://github.com/pandas-dev/pandas/blob/78ec27668163808b3488da6ed2c64d0b8e9bfbc8/pandas/core/indexes/timedeltas.py#L330-L331 for the code where they change freq=None to freq="D"
Should we force the user to do 3 args even if the docs example only uses 2.
Happy to open a PR in the pandas repo to ask for clarification.

Reply:
https://github.com/pandas-dev/pandas/issues/62161

I have asked for clarifications.